astropy/cosmology/io/yaml.py

```diff
@@ -0,0 +1,174 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+The following are private functions, included here **FOR REFERENCE ONLY** since
+the io registry cannot be displayed. These functions are registered into
+:meth:`~astropy.cosmology.Cosmology.to_format` and
+:meth:`~astropy.cosmology.Cosmology.from_format` and should only be accessed
+via these methods.
+"""  # this is shown in the docs.
+
+from astropy.cosmology.core import _COSMOLOGY_CLASSES, Cosmology
+from astropy.cosmology.connect import convert_registry
+from astropy.io.misc.yaml import AstropyDumper, AstropyLoader, dump, load
+
+__all__ = []  # nothing is publicly scoped
+
+
+##############################################################################
+# Serializer Functions
+# these do Cosmology <-> YAML through a modified dictionary representation of
+# the Cosmology object. The Unified-I/O functions are just wrappers to the YAML
+# that calls these functions.
+
+
+def yaml_representer(tag):
+    """:mod:`yaml` representation of |Cosmology| object.
+
+    Parameters
+    ----------
+    tag : str
+        The class tag, e.g. '!astropy.cosmology.LambdaCDM'
+
+    Returns
+    -------
+    representer : callable[[`~astropy.io.misc.yaml.AstropyDumper`, |Cosmology|], str]
+        Function to construct :mod:`yaml` representation of |Cosmology| object.
+    """
+    def representer(dumper, obj):
+        """Cosmology yaml representer function.
+
+        Parameters
+        ----------
+        dumper : `~astropy.io.misc.yaml.AstropyDumper`
+        obj : `~astropy.cosmology.Cosmology`
+
+        Returns
+        -------
+        str
+            :mod:`yaml` representation of |Cosmology| object.
+        """
+        # convert to mapping
+        map = obj.to_format("mapping")
+        # remove the cosmology class info. It's already recorded in `tag`
+        map.pop("cosmology")
+        # make the metadata serializable in a lossless way
```

Comment:
I think the "lossless way" refers to the dict order, since YAML doesn't preserve order for mappings. So say that directly if that is the case.

Reply:
Done!